modules/client-configuration/pages/tshoot-clients.adoc

```diff
@@ -140,9 +140,53 @@ grains:
 
 == Registering Older Clients
 
+////
+LKB 2020-06-03
+
+CCFR for reference:
+
+* Cause: Older clients ({centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients) need to have scp. python, and perl installed, and the server needs to support older types of SSL encryption, in order to successfully register them.
+
+* Consequence: When trying to register on the CLI, registration fails with this error:
++
+----
+Repository '<Repository_Name>' is invalid.
+[|] Valid metadata not found at specified URL(s)
+Please check if the URIs defined for this repository are pointing to a valid repository.
+Skipping repository '<Repository_Name>' because of the above error.
+Download (curl) error for 'www.example.com':
+Error code: Unrecognized error
+Error message: error:1409442E:SSL routines:SSL3_READ_BYTES:tlsv1 alert protocol version
+----
++
+When trying to register on the {webui}, registration fails with this error:
++
+----
+Rendering SLS 'base:bootstrap' failed: Jinja error: >>> No TLS 1.2 and above for RHEL6 and SLES11. Please check your Apache config. <<< Traceback (most recent call last): File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 392, in render_jinja_tmpl output = template.render(**decoded_context) File "/usr/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render return original_render(self, *args, **kwargs) File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render return self.environment.handle_exception(exc_info, True) File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception reraise(exc_type, exc_value, tb) File "/usr/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise raise value.with_traceback(tb) File "<template>", line 53, in top-level template code File "/usr/lib/python3.6/site-packages/salt/utils/jinja.py", line 211, in jinja_raise raise TemplateError(msg) salt.exceptions.TemplateError: >>> No TLS 1.2 and above for RHEL6 and SLES11. Please check your Apache config. <<< ; line 53 --- [...] {%- if not grains['os_family'] == 'Debian' %} {%- set bootstrap_repo_request = salt['http.query'](bootstrap_repo_url + 'repodata/repomd.xml', status=True, verify_ssl=False) %} {# 901 is a special status code for the TLS issue with RHEL6 and SLE11. #} {%- if bootstrap_repo_request['status'] == 901 %} {{ raise(bootstrap_repo_request['error']) }} <====================== {%- endif %} {%- set bootstrap_repo_exists = (0 < bootstrap_repo_request['status'] < 300) %} bootstrap_repo: file.managed: [...] ---
+----
+
+* Fix: Before registering {centos}{nbsp}6, {oracle}{nbsp}6, {rhel}{nbsp}6, or {sleses}{nbsp}6 clients:
++
+* Ensure you have installed python and scp
```

Comment:
No, this is wrong. This is not about old clients. It can happen on any client, and IMHO it should be an admonition for all of them, or a requirement, or something similar.

And if you are going to have it as part of this PR, you should edit the PRs title (but FMPOV you should think about this and do it on a separate PR).

Reply:
oh, you want it as a completely separate troubleshooting section? I can do that Thurs for you :+1:

Reply:
I am not sure this belongs to troubleshooting. Shouldn't we have a "Requirements" page that is common to all OS maybe?

Reply:
There's a new clients hardware requirements table in the Install Guide (modules/installation/pages/client-requirements.adoc), we could add a software pre-req note as well? Otherwise, we could put it at the beginning of the Client Cfg, in modules/client-configuration/pages/supported-features.adoc ?

Reply:
I'd use `modules/installation/pages/client-requirements.adoc`.

Maybe not all customers will read `modules/client-configuration/pages/supported-features.adoc`

Just keep in mind `python` and `scp` are not the package name (those change, are different for different OS). In fact I am not 100% sure about python. I know it's required for salt and salt-ssh, but not so sure about traditional (and I guess python2 or python3 are needed depending on the OS).

Please check with the salt team.

Reply:
OK, I'll do that in a new PR, so that we can get this one in.